File: Verifying an Alien Dictionary.cpp
class Solution {
public:
    bool isAlienSorted(vector<string>& words, string order) {
       map <char,int> temp;
       int j;
        for(int i=0;i<order.size();i++)
        {
            temp[order[i]]=i;
        }

        for( int i=0;i<words.size()-1;i++)
        {
            j=0;
            while(j<words[i].size() && j<words[i+1].size())
            {
                if(temp[words[i][j]] < temp[words[i+1][j]]) 
                break;
                if(temp[words[i][j]] > temp[words[i+1][j]] )
                return false;

                j++;
            }

            if(j==words[i+1].size() && j<words[i].size()) 
            return false; 
        }        

        return true; 
    }   
};

File: week-1/python/verifying-an-alien-dictionary.py
class Solution:
    def isAlienSorted(self, words: List[str], order: str) -> bool:
        temp = {char: i for i, char in enumerate(order)}
        for i in range(len(words) - 1):
            j = 0
            while j < len(words[i]) and j < len(words[i + 1]):
                if temp[words[i][j]] < temp[words[i + 1][j]]:
                    break
                if temp[words[i][j]] > temp[words[i + 1][j]]:
                    return False
                j += 1

            if j == len(words[i + 1]) and j < len(words[i]):
                return False

        return True